File: src/kernellens/domain/verification.py
from dataclasses import dataclass, fields
from enum import StrEnum


class VerificationStatus(StrEnum):
    NOT_RUN = "not_run"
    PASSED = "passed"
    FAILED = "failed"
    INCONCLUSIVE = "inconclusive"


@dataclass(frozen=True)
class VerificationState:
    syntax: VerificationStatus = VerificationStatus.NOT_RUN
    api_evidence: VerificationStatus = VerificationStatus.NOT_RUN
    compilation: VerificationStatus = VerificationStatus.NOT_RUN
    correctness: VerificationStatus = VerificationStatus.NOT_RUN
    performance: VerificationStatus = VerificationStatus.NOT_RUN

    def __post_init__(self) -> None:
        for field in fields(self):
            if not isinstance(getattr(self, field.name), VerificationStatus):
                raise TypeError(f"{field.name} must be a VerificationStatus")

File: src/domain/verification.ts
/**
 * Verification states for kernel code.
 *
 * Each verification dimension is tracked independently.
 * A dimension can be: not_run | passed | failed | inconclusive.
 */

import { z } from 'zod';

export const VerificationResult = z.enum(['not_run', 'passed', 'failed', 'inconclusive']);
export type VerificationResult = z.infer<typeof VerificationResult>;

export const VerificationState = z.object({
  syntax: VerificationResult.default('not_run'),
  api: VerificationResult.default('not_run'),
  compilation: VerificationResult.default('not_run'),
  correctness: VerificationResult.default('not_run'),
  performance: VerificationResult.default('not_run'),
});
export type VerificationState = z.infer<typeof VerificationState>;

/** Create a fresh verification state with all dimensions unset. */
export function createVerificationState(): VerificationState {
  return {
    syntax: 'not_run',
    api: 'not_run',
    compilation: 'not_run',
    correctness: 'not_run',
    performance: 'not_run',
  };
}
